web/packages/selfhosted/test/polyfill/classic_frames_injected/test.ts

```diff
@@ -13,11 +13,28 @@ describe("Flash inside frame with injected ruffle", () => {
     it("polyfills inside a frame", async () => {
         await injectRuffleAndWait(browser);
         await browser.switchToFrame(await browser.$("#test-frame"));
-        await browser.$("<ruffle-object />").waitForExist();
+        // TODO: After https://github.com/webdriverio/webdriverio/issues/13218 is fixed, use browser.$("<ruffle-object />")
+        await browser
+            .$(
+                () =>
+                    (
+                        document.querySelector(
+                            "#test-frame",
+                        ) as HTMLIFrameElement
+                    ).contentDocument?.body.querySelector(
+                        "ruffle-object",
+                    ) as HTMLElement,
+            )
+            .waitForExist();
 
-        const actual = await browser
-            .$("#test-container")
-            .getHTML({ includeSelectorTag: false, pierceShadowRoot: false });
+        // TODO: After https://github.com/webdriverio/webdriverio/issues/13218 is fixed
+        // use browser.$("#test-container").getHTML({ includeSelectorTag: false, pierceShadowRoot: false });
+        const actual = await browser.execute(() => {
+            const el = (
+                document.querySelector("#test-frame") as HTMLIFrameElement
+            ).contentDocument?.body.querySelector("#test-container");
+            return el?.innerHTML;
+        });
```

Comment:
Similar question about these.

Reply:
Probably. Untested, but I guess this may work.

```ts
export async function geContainertHTMLFromFrame(browser: WebdriverIO.Browser) {
        const actual = await browser.execute(() => {
            const el = (
                document.querySelector("#test-frame") as HTMLIFrameElement | HTMLFrameElement
            ).contentDocument?.body.querySelector("#test-container");
            return el?.innerHTML;
        });
        return actual;
}
```

Reply:
It did work indeed, thank you!